geth/transactions/ethtxclient.go

```diff
@@ -84,8 +84,8 @@ func toCallArg(msg ethereum.CallMsg) interface{} {
 	if msg.Value != nil {
 		arg["value"] = (*hexutil.Big)(msg.Value)
 	}
-	if msg.Gas != nil {
-		arg["gas"] = (*hexutil.Big)(msg.Gas)
+	if msg.Gas != 0 {
```

Comment:
Is `0` for sure an invalid value for Gas?
Before the logic assumed that there are 3 types of values: non-set (`nil`), set, but 0, set, != 0.
Here we lost one state.

Reply:
if so I'd extracted it to a constant to simplify changes in the future

Reply:
Hence my question above (https://github.com/status-im/status-go/pull/702#issuecomment-368442046). I'm not against keeping the old version that uses `nil`, tbh.

Reply:
Reverted the changes to `Gas` type. I don't think that we gain much by doing that.